### PR TITLE
Avoid rewriting URLs in the proxy, if the application is proxy-aware.

### PR DESCRIPTION
**Pull request by vishh**
Related to #14624 

cc @lavalamp

### CODE REVIEW

File: pkg/util/proxy/transport.go [hunk 1]
```diff
@@ -136,11 +136,13 @@ func (t *Transport) rewriteURL(targetURL string, sourceURL *url.URL) string {
 	url.Scheme = t.Scheme
 	url.Host = t.Host
 	origPath := url.Path
-	url.Path = path.Join(t.PathPrepend, url.Path)
-
-	if strings.HasSuffix(origPath, "/") {
-		// Add back the trailing slash, which was stripped by path.Join().
-		url.Path += "/"
+	// Do not rewrite URL if the sourceURL already contains the necessary prefix.
+	if !strings.HasPrefix(url.Path, t.PathPrepend) {
```
Comment (lavalamp) [inline pkg/util/proxy/transport.go]: nit: Phrase in positive:

```go
if strings.HasPrefix(url.Path, t.PathPrepend) {
  return url.String()
}
```

Comment (vishh) [inline pkg/util/proxy/transport.go]: Done

**Comment by lavalamp** [comment]
LGTM

**Comment by lavalamp** [comment]
From our chat this morning, I'm guessing you'll probably want to cherry-pick this.

**Comment by vishh** [comment]
Yeah. This one and another patch that updated Grafana configs.

**Comment by bgrant0607** [comment]
See #5897.
cc @enisoc

**Comment by vishh** [comment]
@davidopp: This PR needs to go in soon because it needs to be cherry-picked into v1.1

**Comment by davidopp** [comment]
Tests finally passed, merging manually.

**Pull request merged**